### ISSUE TITLE
Details about how to integrate with x265 encoder 

### ISSUE THREAD

**Issue by aburkleaux-amazon** [DolbyLaboratories/dolby-encoding-engine#1]
Hi,

We're trying to setup the DEE reference encoder implementation using https://bitbucket.org/multicoreware/x265/downloads/ as the x265 encoder.  The paths to the x265 implementation used in the Makefile for this project don't line up with the code in the x265 package.  

Can you provide more details on what version of x265 you are using in the reference implementation?  A link to the specific package that works with dolby-encoding-engine/plugins/code/hevc_enc/x265/make/hevc_enc_x265/linux_amd64_gnu/Makefile would be most helpful.

Thanks!
Alex B

**Comment by mszym1** [comment]
Hi Alex.
Have you seen this readme file? https://github.com/DolbyLaboratories/dolby-encoding-engine/blob/master/plugins/code/hevc_enc/x265/readme.txt

In short, build x265 from bitbucket (we know our plugin works with version 2.4, 2.5 and 2.6 - we did most of testing using 2.5), put required header files and libraries to specified folder structure, set environment variable with its location and then build DEE plugin.

In case of problems, let me know where have you stuck.

**Comment by aburkleaux-amazon** [comment]
Yes, we saw the README, but missed that we needed to copy the files over to that folder structure.  We setup the folders and built the DEE with x265.  

Now we are encountering another problem.  It looks like we get through metadata generation part, but we are failing with an access denied error on the encoder step.  The permissions for last ouput file mentioned in the log  are rwx for the user running the dee command.  We also checked the shared object file for x265 and didn't see any issue there.  


Commandline:
./dee -v ../VanlifeOnBeach_UHD.mxf --progress --xml template_xmls/example.xml --log-file logfile.log --verbose debug

XML: 
https://github.com/DolbyLaboratories/dolby-encoding-engine/blob/master/plugins/code/hevc_enc/x265/example.xml

Logfile attached.
[logfile.log](https://github.com/DolbyLaboratories/dolby-encoding-engine/files/1553684/logfile.log)

**Comment by mszym1** [comment]
I looks like plugin failed while trying to access 10-bit encoder in x265 library.
Have you used multilib.sh script to build x265? This is the way to build 8/10-bit encoder combo, rather than base 8-bit only.

**Comment by aburkleaux-amazon** [comment]
Thanks for the clue.  We had run multlib.sh.  We checked the library we were using in the plugin and found the problem. 

We are were using the x265 shared library in: x265/x265_v2.6/build/linux/libx265.so
The library produced using  multilib.sh that includes the 8/10/12 bit archives is written to: x265/x265_v2.6/build/linux/8bit/libx265.so

We switched to the "8bit" library and our dee job completed successfully.

Thanks!
Alex
